script.js

```diff
@@ -1,4 +1,123 @@
+var player1Score = 0;
+var player2Score = 0;
+//total score for player
+var totalPlayer1Score = [];
+var totalPlayer2Score = [];
+var totalSumPlayer1 = 0;
+var totalSumPlayer2 = 0;
+//dices for players
+var player1Dice1 = 0;
+var player1Dice2 = 0;
+var player2Dice1 = 0;
+var player2Dice2 = 0;
+
+var gameMode = `player one roll`;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == `player one roll`) {
+    return player1Roll();
+  } else if (gameMode == `player one choose`) {
+    return player1Choose(input);
+  } else if (gameMode == `player two roll`) {
+    return player2Roll();
+  } else if (gameMode == `player two choose`) {
+    return player2Choose(input);
+  } else if (gameMode == `winner`) {
+    return showWinner();
+  } else if (gameMode == `leaderboard`) {
+    return showTotalScore();
+  }
+};
+
+var diceRoll = function () {
+  var randomInteger = Math.ceil(Math.random() * 6);
+  return randomInteger;
+};
+
+var inputNumber = function (input) {
+  if (isNaN(input) || input == ``) {
+    return `<div id="instructions">Oops, you did not enter a number. Please enter a valid number (1 or 2).</div>`;
+  } else {
+    //verifies if input is 1 or 2
+    if (input != 1 || input != 2) {
+      return `<div id="instructions">It looks like you didn't enter 1 or 2. Please select either dice 1 or 2.</div>`;
+    }
+  }
+};
+
+var player1Roll = function () {
```

Comment:
Notice how player1Roll and player2Roll have a lot of repeated code in it? That's a good sign that we could  refactor it into a function that can be reused